### PR TITLE
mTLS + authorisation

### PR DESCRIPTION
**Pull request by tcsc**
Implements authentication via mTLS and a simple authorisation policy
(i.e. only the task owner can interact with it).

Also:
 - removes the insecure communication options from both the client and
   server. Using mTLS for authentication & authoristion made them
   largely redundant in any case
 - refactors the secret generation code in the Makefile
 - adds some basic shell scripts to generate client certificates

### CODE REVIEW

File: cmd/levityd/main.go [hunk 1]
```diff
@@ -50,11 +58,45 @@ func expandPaths() error {
...
+	log.Printf("Loading TLS certificate from %s", certificatePath)
+	log.Printf("Loading private key from %s", privateKeyPath)
+	log.Printf("Client CA Cert path: %s", clientCACertPath)
+
+	idCert, err := tls.LoadX509KeyPair(certificatePath, privateKeyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	tlsCfg := &tls.Config{
```
Comment (russjones) [inline cmd/levityd/main.go:78]: Design document says you'll only support TLS 1.3, but I think this supports TLS 1.2 and 1.3.

**Review by russjones — APPROVED** [review]

File: cmd/levity/main.go [hunk 2]
```diff
@@ -40,9 +41,12 @@ func init() {
 	flags.DurationVarP(&timeout, "timeout", "t", timeout,
 		"Timeout for GRPC requests")
 
-	flags.StringVar(&caCertPath, "ca", "", "Override CA file")
-	flags.BoolVar(&connectInsecure, "insecure", false,
-		"Connect insecurely to the server (i.e. no TLS)")
+	flags.StringVar(&caCertPath, "ca", "", "Override system CA with provided CA")
+
+	flags.StringVarP(&idPrivateKey, "key", "k", "./certs/usr-key.pem",
```
Comment (russjones) [inline cmd/levity/main.go]: This is different than the directory for the server, I think that expects certs to be in `./cert`. Also, this expects the certs to be in `bin` but the certs are placed in `repoRoot/certs`.

Comment (tcsc) [reply cmd/levity/main.go]: I've removed the defaults for these options - they are now required arguments on CLI. I found the defaults less useful than I'd hoped, in any case.

File: cmd/levityd/main.go (hunk 1 above)
Comment (tcsc) [reply cmd/levityd/main.go:78]: Added minimum version to TLS config, and integration test to assert it is enforced.

**Pull request merged**